x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/utils/ExceptionsHelper.java

```diff
@@ -112,7 +113,29 @@ public static String shardFailuresToErrorMsg(String jobId, ShardSearchFailure[]
         return requireNonNull(obj, paramName.getPreferredName());
     }
 
+    /**
+     * Unwrap the exception stack and return the most likely cause.
+     * This method has special handling for {@link SearchPhaseExecutionException}
+     * where it returns the cause of the first shard failure.
+     *
+     * @param t raw Throwable
+     * @return unwrapped throwable if possible
+     */
     public static Throwable unwrapCause(Throwable t) {
```

Comment:
I think this is a bad name as it collides with `org.elasticsearch.ExceptionsHelper.unwrapCause`

Not saying `getRootCauseException` is a good one, maybe something that indicates the intent: unwrapping _search responses_, e.g. `unwrapSearchException` or `findSearchExceptionRootCause`.

I think this method should _not_ become a general exception handling method, but _only_ for handling search errors. Otherwise the chances are high that code sharing lead to undesired bugs.

For that reason I think it would be good to move it out of the ml subpackage (despite this is just a logical but not technically limiting) into a more common area.

Transform should use this without indirection.

Reply:
It is basically an `unwrapCause` that also handles `SearchPhaseExecutionException`. The problem with `SearchPhaseExecutionException` is that it doesn't implement `ElasticsearchWrapperException`. Let's discuss this further offline.

Reply:
OK, I have refactored the code to keep a method named `findSearchExceptionRootCause` instead of replacing `unwrapCause`.